crawler/crawler.go

```diff
@@ -6,60 +6,28 @@ import (
 	"sort"
 )
 
-// CrawlerInterface defines the interface for a web crawler.
-type CrawlerInterface interface {
-	Run()
-	GetResult() []string
-	SortLinks()
-}
-
 // Strategy represents a web crawling strategy.
 type Strategy interface {
 	Run(url *url.URL) []string
 }
 
-// Crawler represents a web crawler.
-type Crawler struct {
-	Strategy Strategy // Algorithm to use
-	Result   []string // Result of the crawl
-}
-
-// NewCrawler creates a new web crawler with the specified strategy.
-func NewCrawler(strategy string) (*Crawler, error) {
+// Run starts the web crawling process with the specified root URL.
+// It returns the result of the crawl or an error if any occurred.
+func Run(rootUrl string, strategy string) ([]string, error) {
 	// Create the strategy
-	alg, err := createStrategy(strategy)
+	st, err := createStrategy(strategy)
 	if err != nil {
 		return nil, err
 	}
-	// Create the crawler
-	crawler := &Crawler{
-		Strategy: alg,
-		Result:   []string{},
-	}
-	return crawler, nil
-}
-
-// Run starts the web crawling process with the specified root URL.
-// It returns the result of the crawl or an error if any occurred.
-func (c *Crawler) Run(rootUrl string) ([]string, error) {
 	// Parse the given URL
 	parsedURL, err := url.Parse(rootUrl)
 	if err != nil {
-		return []string{}, errors.New("error parsing URL")
+		return nil, err
```

Comment:
Both options are correct the old one and the new one. The new one might give a more specific description of the issue while parsing however the the old one is user friendlier in the case then en user is not a developer.